### ISSUE TITLE
read snapshots? [JIRA: RIAK-2321]

### ISSUE THREAD

**Issue by benoitc** [basho/eleveldb#115]
Reading the code it seems that read snapshots [1] are not supported. Did I miss something? Do you plan to add such support?

[1] http://leveldb.googlecode.com/svn/trunk/doc/index.html

**Comment by slfritchie** [comment]
Added 2.1 milestone, subject to further planning review.

**Comment by benoitc** [comment]
Looks like https://github.com/basho/eleveldb/compare/arg-snapshot-2 is implementing it. I wonder if anyone tried it.

**Comment by benoitc** [comment]
I would like to relaunch the discussion here. What I would like is having a way to read concurrently from writing. So eventually read could be done in another process. Is there a way to do that right now? Does the iterator provides a snapshot view of the db?

**Comment by Basho-JIRA** [comment]
Yes.  Iterator is a snapshot view.  Iterator, once constructed, can Seek to closest key as well as traverse Next and Prev. 

**_[posted via JIRA by Matthew Von-Maszewski]**_

**Comment by benoitc** [comment]
can the iterator be passed to another erlang process ?

**Comment by Basho-JIRA** [comment]
Erlang process on same machine, yes.  Linux process no. 

**_[posted via JIRA by Matthew Von-Maszewski]**_

**Comment by benoitc** [comment]
Yes I meant  , opening an iterator and use the ref in another process that will eventually close it. would it work?
